ast2vec/cloning.py

```diff
@@ -0,0 +1,213 @@
+import argparse
+import errno
+from functools import partial
+import json
+import logging
+from multiprocessing.dummy import Pool
+import os
+import shutil
+import subprocess
+import zlib
+
+
+class LinguistFailedError(Exception):
+    """
+    Raised when we fail to classify the source files.
+    """
+    pass
+
+
+class RepoCloner:
+    """
+    Clone repositories from provided urls / files with urls.
+    Use enry to classify files and delete redundant files if needed.
+    """
+    def __init__(self, linguist, redownload, languages=None,
+                 log_level=logging.INFO, num_threads=1):
+        self._log = logging.getLogger("repo_cloner")
+        self._log.setLevel(log_level)
+        self._languages = languages
+        self._linguist, self._is_enry = self._find_enry(linguist)
+        self._num_threads = num_threads
+        self._redownload = redownload
+
+    def clone_repo(self, git_url, ignore, target_dir):
+        """
+        Clone repository into a separate folder inside of target directory.
+
+        :param git_url: Url of Git repository.
+        :param ignore: Flag for ignoring Git failures.
+        :param target_dir: Target directory. New folder will be created inside of target_dir.
+        :return: Path to downloaded Git repository.
+        """
+        git_url = self._prepare_repo_url(git_url)
+
+        try:
+            repo_dir = self._prepare_repo_dir(git_url, target_dir, self._redownload)
+        except FileExistsError as e:
+            self._log.warning("%s already cloned, skipping.", git_url)
+            return
+
+        env = os.environ.copy()
+        env["GIT_TERMINAL_PROMPT"] = "0"
```

Comment:
It's better to put explanation why some magic happens with env variable)
Like: "force git not to ask anything".